### DISCUSSION TITLE
Problems with Docker installation - Migration fails with PK errors

### DISCUSSION THREAD

**Discussion by RicardoFernandez-UY** [FlowiseAI/Flowise#1950] [Q&A] [answered]
Hi,
I'm using Flowise 1.6.1 on Docker with a Postgres 16.2 database also on Docker.
When I run the Flowise container for the first time it starts but also shows these errors regarding already existing PKs when initializing the database. 
The database is newly created so it is completely empty before the initialization process.

```
2024-03-14 13:43:09 [INFO]: ⚡️ [server]: Flowise Server is listening at 3033
2024-03-14 13:43:09 [INFO]: 📦 [server]: Data Source is being initialized!
Migration "AddFeedback1707213601923" failed, error: relation "PK_98419043dd704f54-9830ab78f8" already exists
2024-03-14 13:43:09 [ERROR]: ❌ [server]: Error during Data Source initialization: relation "PK_98419043dd704f54-9830ab78f8" already exists
QueryFailedError: relation "PK_98419043dd704f54-9830ab78f8" already exists
    at PostgresQueryRunner.query (/usr/local/lib/node_modules/flowise/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:219:19)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async AddFeedback1707213601923.up (/usr/local/lib/node_modules/flowise/dist/database/migrations/postgres/1707213601923-AddFeedback.js:6:9)
    at async MigrationExecutor.executePendingMigrations (/usr/local/lib/node_modules/flowise/node_modules/typeorm/migration/MigrationExecutor.js:225:17)
    at async DataSource.runMigrations (/usr/local/lib/node_modules/flowise/node_modules/typeorm/data-source/DataSource.js:265:35)
    at async /usr/local/lib/node_modules/flowise/dist/index.js:77:13
```

The container starts and I can login, but once I start using Flowise many things do not work (probably because of these migration errors)

Thank you for your help!

**Comment by amansoniamazatic** [comment]
@RicardoFernandez-UY  Use below Dockerfile:

```
# Use Node.js 18 Alpine as base image
FROM node:18-alpine

# Install PNPM globally
RUN npm i -g pnpm

# Create the /data directory with appropriate permissions
RUN mkdir -p /data && chmod -R 777 /data

# Create the flowise directory with appropriate permissions
RUN mkdir -p /usr/local/lib/node_modules/flowise && chmod -R 777 /usr/local/lib/node_modules/flowise

# Switch to a non-root user for security
USER node

# Arguments that can be passed at build time
ARG FLOWISE_PATH=/usr/local/lib/node_modules/flowise
ARG BASE_PATH=/data/.flowise
ARG DATABASE_PATH=$BASE_PATH
ARG APIKEY_PATH=$BASE_PATH
ARG SECRETKEY_PATH=$BASE_PATH
ARG LOG_PATH=$BASE_PATH/logs

# Set the ARGs as ENV
ENV FLOWISE_PATH=$FLOWISE_PATH
ENV BASE_PATH=$BASE_PATH
ENV DATABASE_PATH=$DATABASE_PATH
ENV APIKEY_PATH=$APIKEY_PATH
ENV SECRETKEY_PATH=$SECRETKEY_PATH
ENV LOG_PATH=$LOG_PATH

# Install dependencies
USER root
RUN apk add --no-cache git python3 py3-pip make g++ build-base cairo-dev pango-dev chromium
USER node

ENV PUPPETEER_SKIP_DOWNLOAD=true
ENV PUPPETEER_EXECUTABLE_PATH=/usr/bin/chromium-browser

# Create the /app directory with appropriate permissions
RUN mkdir -p /home/node/app && chmod -R 777 /home/node/app
WORKDIR /home/node/app

# Clone the Flowise repository from GitHub
RUN git clone https://github.com/FlowiseAI/Flowise.git .

# Install Flowise dependencies
RUN pnpm install
RUN pnpm build

# Set the entry point to run Flowise
CMD ["sh", "-c", "mkdir -p $LOG_PATH && chmod -R 777 $LOG_PATH && mkdir -p $FLOWISE_PATH/uploads && chmod -R 777 $FLOWISE_PATH/uploads && pnpm start"]

```

**Reply by RicardoFernandez-UY** [reply]
Hi @amansoniamazatic 
Thank you for your answer!
I built the image you suggested with the docker compose, created the container, and the error remains exactly the same. And I mean exactly as in character by character.
I've noticed that the image generated is stil :latest (same as when you get it directly from dockerhub). I don't know if that is relevant.
Any other idea will be much appreciated!

This is the docker compose run:
`root@vmi1100571:/Flowise/Flowise/docker# docker compose up -d
WARN[0000] The "CORS_ORIGINS" variable is not set. Defaulting to a blank string.
WARN[0000] The "IFRAME_ORIGINS" variable is not set. Defaulting to a blank string.
WARN[0000] The "FLOWISE_FILE_SIZE_LIMIT" variable is not set. Defaulting to a blank string.
WARN[0000] The "DEBUG" variable is not set. Defaulting to a blank string.
WARN[0000] The "DATABASE_SSL" variable is not set. Defaulting to a blank string.
WARN[0000] The "DATABASE_SSL_KEY_BASE64" variable is not set. Defaulting to a blank string.
WARN[0000] The "FLOWISE_SECRETKEY_OVERWRITE" variable is not set. Defaulting to a blank string.
WARN[0000] The "LOG_LEVEL" variable is not set. Defaulting to a blank string.
WARN[0000] The "DISABLE_FLOWISE_TELEMETRY" variable is not set. Defaulting to a blank string.
[+] Building 314.4s (11/11) FINISHED                                                                                       docker:default
 => [flowise internal] load build definition from Dockerfile                                                                         0.0s
 => => transferring dockerfile: 480B                                                                                                 0.0s
 => [flowise internal] load metadata for docker.io/library/node:18-alpine                                                            0.7s
 => [flowise internal] load .dockerignore                                                                                            0.0s
 => => transferring context: 2B                                                                                                      0.0s
 => [flowise 1/7] FROM docker.io/library/node:18-alpine@sha256:ca9f6cb0466f9638e59e0c249d335a07c867cd50c429b5c7830dda1bed584649      2.9s
 => => resolve docker.io/library/node:18-alpine@sha256:ca9f6cb0466f9638e59e0c249d335a07c867cd50c429b5c7830dda1bed584649              0.0s
 => => sha256:e7ced292c644a1f7bc776dcc401164b67c9224f8592cc83b8c42e237668a0c7f 40.25MB / 40.25MB                                     0.7s
 => => sha256:b32c0114bba5af3e85af37dbc23b1e026850aba590099b81bf75946327b3a9e8 2.34MB / 2.34MB                                       0.3s
 => => sha256:f3748d9674b0ca905fe23e1cb4ad0e49d6a605dbbfb9d0cf485f300a03f1eeff 450B / 450B                                           0.1s
 => => sha256:ca9f6cb0466f9638e59e0c249d335a07c867cd50c429b5c7830dda1bed584649 1.43kB / 1.43kB                                       0.0s
 => => sha256:affdf979bd8ec516bf189d451b8ac68dd50adc49adc4c4014963556c11efeda4 1.16kB / 1.16kB                                       0.0s
 => => sha256:24d8fcd7167fb06e91dc7228311105013dc042f6875ff2528ff7a41c04770112 7.14kB / 7.14kB                                       0.0s
 => => extracting sha256:e7ced292c644a1f7bc776dcc401164b67c9224f8592cc83b8c42e237668a0c7f                                            1.9s
 => => extracting sha256:b32c0114bba5af3e85af37dbc23b1e026850aba590099b81bf75946327b3a9e8                                            0.1s
 => => extracting sha256:f3748d9674b0ca905fe23e1cb4ad0e49d6a605dbbfb9d0cf485f300a03f1eeff                                            0.0s
 => [flowise 2/7] RUN apk add --no-cache git                                                                                         1.9s
 => [flowise 3/7] RUN apk add --no-cache python3 py3-pip make g++                                                                    4.5s
 => [flowise 4/7] RUN apk add --no-cache build-base cairo-dev pango-dev                                                              2.9s
 => [flowise 5/7] RUN apk add --no-cache chromium                                                                                    8.5s
 => [flowise 6/7] RUN npm install -g flowise                                                                                       270.2s
 => [flowise 7/7] WORKDIR /data                                                                                                      0.0s
 => [flowise] exporting to image                                                                                                    22.6s
 => => exporting layers                                                                                                             22.6s
 => => writing image sha256:5d75bc2a20d62afe7eac5d50e846985194ca1498e4dae14f854519c9d8c9462f                                         0.0s
 => => naming to docker.io/library/docker-flowise                                                                                    0.0s
[+] Running 1/1
 ✔ Container docker-flowise-1  Started                     `

**Comment by Niltoncota** [comment]
I'm having the same problem. I uninstalled and reinstalled several times and the problem remains the same.

**Comment by amansoniamazatic** [comment, marked as answer]
@RicardoFernandez-UY @Niltoncota Please install a slightly older version, like v1.5.1 or v1.6.0. I think the feedback option is a new feature in the chatbot and it is not properly executed in v1.6.1, which is causing errors.

**Reply by RicardoFernandez-UY** [reply]
Hi @Niltoncota 
Thank you very much! With older versions it works! I tried with 1.6.0 and it is up and running.

**Reply by amansoniamazatic** [reply]
You're welcome!
